test_utils.sh

```diff
@@ -44,16 +50,20 @@ authorize() {
     local policies=$2
     local entities=$3
     local schema=$4
+    local links=$5
     echo " Running authorization on ${policies}"
     for decision in ALLOW DENY
     do
         for file in "$folder/$decision"/*.json
         do
-            if [ -z "$schema" ]
+            if [ -z "$schema" -a -z "$links" ]
             then
                 IFS=$'\n' read -r -d '' -a tmp_array < <(cedar authorize --policies "$folder/$policies"  --entities "$folder/$entities" --request-json "$file" -v && printf '\0')
-            else
+            elif [ -z "$links" ]
+            then
                 IFS=$'\n' read -r -d '' -a tmp_array < <(cedar authorize --policies "$folder/$policies" --schema "$folder/$schema" --schema-format human --entities "$folder/$entities" --request-json "$file" -v && printf '\0')
+            else
+                IFS=$'\n' read -r -d '' -a tmp_array < <(cedar authorize --policies "$folder/$policies" -k "$folder/$links" --schema "$folder/$schema" --schema-format human --entities "$folder/$entities" --request-json "$file" -v && printf '\0')
             fi
```

Comment:
are we missing a case where `-z "$schema"` but not `-z "$links"`?  just doesn't come up in the current tests?

Might be clearer (and less duplication) to define something like `$schema_arg` to be either `--schema "$folder/$schema" --schema-format human` or empty, and likewise `$links_arg`, and then we can have the command just once and reference `$schema_arg` and `$links_arg`

Reply:
> are we missing a case 

No because that situation cannot happen: Args come in by order with schema first, links second. So if links is there, then schema must be too.

These are bash functions so I don't know how to make the args by name.

Reply:
Got it, seems good.

> I don't know how to make the args by name

I was meaning something like

```
if [ -z "$schema" ] then schema_arg='' else schema_arg='--schema "$folder/$schema" --schema-format human' fi
if [-z "$links" ] then links_arg='' else links_arg='-k "$folder/$links"' fi
IFS=$'\n' read -r -d '' -a tmp_array < <(cedar authorize ... $schema_arg $links_arg ... -v && printf '\0')
```

Reply:
Ok. I have stacked commits with other PRs, so I'll fix it after all those are in.